src/mito2/src/manifest/action.rs

```diff
@@ -45,6 +45,8 @@ pub enum RegionMetaAction {
 pub struct RegionChange {
     /// The metadata after changed.
     pub metadata: RegionMetadataRef,
+    #[serde(default, with = "humantime_serde")]
+    pub ttl: Option<Duration>,
```

Comment:
We don't need to persist this as we already store table options in meta server and we only need to update the in memory state of the region.

Reply:
@evenyag @WenyXu This approach is not viable.
When datanode starts, it only scans all regions it owns, which is a set of `DatanodeTableValue`.
https://github.com/GreptimeTeam/greptimedb/blob/6e776d5f988c54ce8481064c5179c387985ec6cd/src/datanode/src/datanode.rs#L213

The name `DatanodeTableValue` is misleading, in fact, it only contains the region-related info:

https://github.com/GreptimeTeam/greptimedb/blob/e328c7067ce95f042b273030dde6875b643934e7/src/common/meta/src/key/datanode_table.rs#L117-L123

So if you need table metadata here, you can only do another lookup according to the table id, for every region.

So simply store the option in table metadata does not work, we have two other approaches:
- A: Write the TTL to region metadata in the alter procedure
- B: Write the TTL to some place that datanode can access during startup, which is manifest

For A, we need to introduce another step to alter region metadata, and I can't see clear benefit.

Reply:
RegionInfo contains a field `region_options`. @WenyXu Does it contain all the table options we need?

Reply:
The proposed changes are included in https://github.com/GreptimeTeam/greptimedb/pull/4848/commits/9df3a4a6218ad03e9e13c65df8674de282484773 and https://github.com/GreptimeTeam/greptimedb/pull/4848/commits/1f8297f6ec9706bbfb60bdba41b1e79cd9516827